backend/src/crd/kubernetes/v2beta1/pipelineversion_types.go

```diff
@@ -0,0 +1,230 @@
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/kubeflow/pipelines/api/v2alpha1/go/pipelinespec"
+	"github.com/kubeflow/pipelines/backend/src/apiserver/model"
+	"gopkg.in/yaml.v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// PipelineVersionSpec defines the desired state of PipelineVersion.
+type PipelineVersionSpec struct {
+	Description   string `json:"description,omitempty"`
+	CodeSourceURL string `json:"codeSourceURL,omitempty"`
+	PipelineName  string `json:"pipelineName,omitempty"`
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Schemaless
+	// +kubebuilder:pruning:PreserveUnknownFields
+	PipelineSpec PipelineIRSpec `json:"pipelineSpec"`
+}
+
+func ToPipelineSpec(p *PipelineIRSpec) (*pipelinespec.PipelineSpec, error) {
+	pDefBytes, err := json.Marshal(p.Value)
+	if err != nil {
+		return nil, err
+	}
+
+	rv := &pipelinespec.PipelineSpec{}
+	err = json.Unmarshal(pDefBytes, rv)
+	if err != nil {
+		return nil, err
+	}
+
+	return rv, nil
+}
+
+// SimplifiedCondition is a metav1.Condition without lastTransitionTime since the database model doesn't have such
+// a concept and it allows a default status in the CRD without a controller setting it.
+type SimplifiedCondition struct {
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])$`
+	// +kubebuilder:validation:MaxLength=316
+	Type string `json:"type" protobuf:"bytes,1,opt,name=type"`
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=True;False;Unknown
+	Status metav1.ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status"`
+	Reason string                 `json:"reason" protobuf:"bytes,5,opt,name=reason"`
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=32768
+	Message string `json:"message" protobuf:"bytes,6,opt,name=message"`
+}
+
+// PipelineVersionStatus defines the observed state of PipelineVersion.
+type PipelineVersionStatus struct {
+	Conditions []SimplifiedCondition `json:"conditions,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// PipelineVersion is the Schema for the pipelineversions API.
+type PipelineVersion struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PipelineVersionSpec   `json:"spec,omitempty"`
+	Status PipelineVersionStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// PipelineVersionList contains a list of PipelineVersion.
+type PipelineVersionList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []PipelineVersion `json:"items"`
+}
+
+func FromPipelineVersionModel(pipeline model.Pipeline, pipelineVersion model.PipelineVersion) (*PipelineVersion, error) {
```

Comment:
Optional: It'd be nice to have these unit tested but it's okay to cover these later through integration tests in the new pipeline store PR.

Reply:
I'd prefer cover this when the pipeline store is implemented if it's fine.